api-server/__test__/lib/server.test.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+const { server } = require('../../lib/server.js');
+const supertest = require('supertest');
+const mockRequest = supertest(server);
+
+describe('web server', () => {
+
+  it('should respond with a 500 on an error', () => {
+
+    return mockRequest
+      .get('/bad')
+      .then(results => {
```

Comment:
you don't need to have the promises for the tests(.then and .catch). For line 12 try with put.